lang/py/src/avro/schema.py

```diff
@@ -315,14 +315,13 @@ def name_ref(self, names):
   namespace = property(lambda self: self.get_prop('namespace'))
   fullname = property(lambda self: self._fullname)
 
+#
 # Logical type class
-class LogicalSchema(Schema):
-  def __init__(self, type, logical_type, other_props=None):
-    self.logical_type = logical_type
+#
 
-    # Call parent ctor
-    Schema.__init__(self, type, other_props)
-    self.set_prop('logicalType', logical_type)
+class LogicalSchema():
+  def __init__(self, logical_type):
+    self.logical_type = logical_type
 
   def validate(self, validation_props=None):
```

Comment:
I think we don't need it, and this is something we can leave it to the derived class to decide what they want to do.